_test/test_utilities_herbert/test_str_to_iarray.m

```diff
@@ -0,0 +1,216 @@
+            func = @()str_to_iarray (' 3 - 14');
+            ME = assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+            assertTrue(contains(ME.message, ...
+                'Invalid format array descriptor, or Inf or NaN found in'));
+        end
+        
+        %--------------------------------------------------------------------------
+        % Multiple tokens
+        %--------------------------------------------------------------------------
+        function test_two_numbers (~)
+            iarr = str_to_iarray ('45,46');
+            assertEqual (iarr, [45,46])
+        end
+        
+        function test_mixed_format_ranges (~)
+            iarr = str_to_iarray ('14:-2:5, 19 -3--6  4:6');
+            assertEqual (iarr, [14:-2:5, 19, -3:-1:-6, 4:6])
+        end
+        
+        %--------------------------------------------------------------------------
+        % Enclosing brackets
+        %--------------------------------------------------------------------------
+        function test_array_scalar (~)
+            iarr = str_to_iarray ('[37]');
+            assertEqual (iarr, 37)
+        end
+        
+        function test_array_range (~)
+            iarr = str_to_iarray (' [ 46,  45] ');
+            assertEqual (iarr, [46,45])
+        end
+        
+        %--------------------------------------------------------------------------
+        % Comment lines, multiple lines
+        %--------------------------------------------------------------------------
+        function test_comment_line (~)
+            iarr = str_to_iarray ('% 35');
+            assertEqual (iarr, NaN(1,0))
+        end
+        
+        function test_trailing_comment (~)
+            iarr = str_to_iarray (' [ 46,  45] % a comment');
+            assertEqual (iarr, [46,45])
+        end
+        
+        function test_two_lines (~)
+            iarr = str_to_iarray ({'45:47','49'});
+            assertEqual (iarr, [45:47,49])
+        end
+        
+        %--------------------------------------------------------------------------
+        % Text other than a single character vector
+        %--------------------------------------------------------------------------
+        function test_characterArray (~)
+            iarr = str_to_iarray (['45,46                  '; ...
+                '99:-1:90  % Hello there']);
+            assertEqual (iarr, [45,46,99:-1:90])
+        end
+        
+        function test_cellArray (~)
+            iarr = str_to_iarray ({'45,46'; '99:-1:90  % Hello there'});
+            assertEqual (iarr, [45,46,99:-1:90])
+        end
+        
+        function test_MatlabString (~)
+            iarr = str_to_iarray ("45,46");
+            assertEqual (iarr, [45,46])
+        end
+        
+        function test_MatlabStringArray (~)
+            iarr = str_to_iarray (["45,46"; "99:-1:90  % Hello there"]);
+            assertEqual (iarr, [45,46,99:-1:90])
```

Comment:
The motivation of the original function was to read a bunch of lines from a text file as a single integer array. It goes back to code I wrote for HET in about 1990 - there's an original Fortran routine that I wrote with the same name...). Accordingly, if fed a character vector, a 2D character array, or (since R2017a) a Matlab string array str_to_iarray returna a single row vector. The str_to_iarray documentation describes the behaviour.